module/VuFind/src/VuFind/ILS/Driver/KohaRest.php

```diff
@@ -2435,9 +2435,18 @@ protected function holdError($error)
      *
      * @return string
      */
-    protected function mapRenewalBlockReason($reason)
+    protected function mapRenewalBlockReason($reason, $itype)
     {
-        return $this->renewalBlockMappings[$reason] ?? 'renew_item_no';
+        if (isset($this->config['ItypeRenewalBlockMappings'][$itype])) {
```

Comment:
With the configuration in array format this can be simplified quite a bit to something like:
```
return $this->config['ItypeRenewalBlockMappings'][$itype][$reason]
    ?? $this->renewalBlockMappings[$reason] 
    ?? 'renew_item_no';
```